docs/design.md

```diff
@@ -91,97 +107,160 @@ This pattern is inspired by [sickle_ui](https://github.com/UmbraLuminosa/sickle_
-            (SpriteKey::Powerup, asset_server.load("powerup.png")),
-        ]
-        .into()
+
+        let files = [
+            ImageHandles::KEY_PLAYER,
+            ImageHandles::KEY_ENEMY,
+            ImageHandles::KEY_POWERUP,
+        ];
+        let map = files
+            .into_iter()
+            .map(|file| (file.to_string(), asset_server.load(file)))
+            .collect();
+
+        Self(map)
     }
 }
 ```
 
-Then set up preloading in a plugin:
+Then start preloading in the `assets::plugin`:
 
 ```rust
-app.register_type::<HandleMap<SpriteKey>>();
-app.init_resource::<HandleMap<SpriteKey>>();
+pub(super) fn plugin(app: &mut App) {
+    app.register_type::<ImageHandles>();
+    app.init_resource::<ImageHandles>();
+}
+```
+
+And finally add a loading check to the `screens::loading::plugin`:
+
+```rust
+fn all_assets_loaded(
+    image_handles: Res<ImageHandles>,
+) -> bool {
+    image_handles.all_loaded(&asset_server)
+}
 ```
 
 ### Reasoning
 
 This pattern is inspired by [bevy_asset_loader](https://github.com/NiklasEi/bevy_asset_loader).
 By preloading your assets, you can avoid hitches during gameplay.
+We start loading as soon as the app starts and wait for all assets to be loaded in the loading screen.
 
-Using an enum to represent your assets encapsulates their file path from the rest of the game code,
-and gives you access to static tooling like renaming in an IDE, and compile errors for an invalid name.
+By using strings as keys, you can dynamically load assets based on input data such as a level file.
+If you prefer a purely static approach, you can also use an `enum YourAssetHandleKey` and `impl AsRef<str> for YourAssetHandleKey`.
+You can also mix the dynamic and static approach according to your needs.
 
-## Spawning
+## Spawn Commands
 
 ### Pattern
 
-Spawn a game object by using an observer:
+Spawn a game object by using a custom command. Inside the command,
+run the spawning code with `world.run_system_once` or  `world.run_system_once_with`:
 
 ```rust
 // monster.rs
-use bevy::prelude::*;
 
-pub(super) fn plugin(app: &mut App) {
-    app.observe(on_spawn_monster);
+#[derive(Debug)]
+pub struct SpawnMonster {
+    pub health: u32,
+    pub transform: Transform,
 }
 
-#[derive(Event, Debug)]
-pub struct SpawnMonster;
+impl Command for SpawnMonster {
+    fn apply(self, world: &mut World) {
+        world.run_system_once_with(self, spawn_monster);
+    }
+}
 
-fn on_spawn_monster(
-    _trigger: Trigger<SpawnMonster>,
+fn spawn_monster(
+    spawn_monster: In<SpawnMonster>,
     mut commands: Commands,
 ) {
     commands.spawn((
         Name::new("Monster"),
+        Health::new(spawn_monster.health),
+        SpatialBundle::from_transform(spawn_monster.transform),
         // other components
     ));
 }
 ```
 
-And then, somewhere else in your code, trigger the observer:
+And then to use a spawn command, add it to `Commands`:
 
 ```rust
+// dangerous_forest.rs
+
 fn spawn_monster(mut commands: Commands) {
```

Comment:
Maybe a little confusing that this function has the same name as the earlier one? Maybe rename it to like `spawn_forest_monster`

Reply:
Very good point!
[`b4d97f9` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/b4d97f962038bb90a8d9dfbcde1a7022e9651c5a)